packages/build-utils/src/lambda.ts

```diff
@@ -102,6 +107,17 @@ export class Lambda {
     assert(typeof runtime === 'string', '"runtime" is not a string');
     assert(typeof environment === 'object', '"environment" is not an object');
 
+    if (architecture !== undefined) {
+      assert(
```

Comment:
You can remove the assert for string if you're going to check the actual values below